### ISSUE TITLE
Ａ列車で行こう9 (A-Train 9) Rendering issues in distant terrain

### ISSUE THREAD

**Issue by Dasug** [doitsujin/dxvk#2626]
Distant (flat) terrain fails to render both in-game as well as in the main menu. Zooming closer to the respective terrain makes it appear again but might cause other terrain to disappear. See this screenshot of the main menu, the pale blue hole in the landscape is not supposed to appear like that:

![A-Train 9 Rendering Bug](https://user-images.githubusercontent.com/8014546/164345212-c524f085-4cb5-40b8-b687-72fe81544e84.png)

This is how the same scene in the main menu is rendered when running just with wine-7.6 (Staging) or on Windows:

![A-Train 9 no bug](https://user-images.githubusercontent.com/8014546/164345292-0a8ebb83-a827-464b-9b22-578ddb9bbe5b.png)

### Software information
Ａ列車で行こう9 version 5.00  Build 4632

I'm using a mod to translate the interface in the above screenshots as the game itself does not have an English language option, however the issue also appears in an unmodified copy of the game.

A demo version of the game is available at https://www.artdink.co.jp/japanese/title/a9/SetupA9v5Viewer.exe which also experiences the rendering issue.

### System information
- GPU: AMD Radeon Vega 64
- Driver: amdgpu / MESA 22.0.1
- Wine version: wine-5.12-14175-gcd0f2d41c2d (Staging) (from Wine-GE-Proton7-8), also tested with wine-7.6 (Staging)
- DXVK version: Tested with v1.10-2-g809b446f and v1.10.1-36-g3004026d (``master`` at the time of writing)

### Apitrace file(s)
- https://drive.google.com/file/d/1Ys_-m_1fZviEQoQJoKpRPixigITR-bi4/view?usp=sharing (created on windows)

### Log files
- d3d9.log: [DXVK v1.10](https://github.com/doitsujin/dxvk/files/8526275/ATrain9v5g_x64_d3d9.log) / [DXVK master](https://github.com/doitsujin/dxvk/files/8526267/ATrain9v5g_x64_d3d9_master.log)
- d3d11.log: -
- dxgi.log: [DXVK v1.10](https://github.com/doitsujin/dxvk/files/8526303/ATrain9v5g_x64_dxgi.log) / [DXVK master](https://github.com/doitsujin/dxvk/files/8526304/ATrain9v5g_x64_dxgi_master.log)

**Comment by Blisto91** [comment]
Yes i see it in the trace. This is on my R9 380
Same ish color or black depending on if `d3d9.floatEmulation` is on or off.


<details>
  <summary>floatEmulation False</summary>
  
![Screenshot_20220616_231608](https://user-images.githubusercontent.com/47954800/174166327-53664351-a208-43af-b8e9-a1a1ab9f085e.png)
</details>

<details>
  <summary>floatEmulation True/Strict</summary>
  
![Screenshot_20220616_231503](https://user-images.githubusercontent.com/47954800/174166469-c4e7caf8-e9e2-4973-bf81-68e19c076a68.png)
</details>

Didn't find that any options fixed it. 
Also saw the same when i tried dxvk 1.5 as a regression test.

**Comment by Dasug** [comment]
Just dropping by again to add that the issue is still present as of dxvk 2.3 and hasn't magically been fixed by another update (There's always hope 😄). Also it seems that the same issue is present in the steam games [A-Train 9 V4.0 : Japan Rail Simulator](https://steamdb.info/app/278550/), [A-Train 9 V3.0 : Railway Simulator](https://steamdb.info/app/324280/) and [A-Train 9](https://steamdb.info/app/285400/), which makes sense as these are all older translated versions of Ａ列車で行こう9.

**Comment by Blisto91** [comment]
Thanks for the retest. I also check the apitrace from time to time to see if it has been fixed with other fixes.

**Comment by WenSimEHRP** [comment]
This issue it not yet fixed unfortunately. I was playing the same game with the latest version of dxvk on Windows and experienced the same problem, while not adding the extra dlls gives me the correct graphics but worse performance.

**Comment by Dasug** [comment]
This issue also affects the (at least outside of Japan) newly released [A-Train 9 V5.0 : Japan Train Simulator Ultimate Edition](https://store.steampowered.com/app/3106160/ATrain_9_V50__Japan_Train_Simulator_Ultimate_Edition/) as this is yet another version of A-Train 9.

**Comment by Blisto91** [comment]
This should be fixed when the new dxbc-spirv shader compiler gets enabled by default for d3d9

**Comment by Blisto91** [comment]
@Dasug can you verify with this build? https://github.com/doitsujin/dxvk/actions/runs/26543903902

**Comment by Dasug** [comment]
> [@Dasug](https://github.com/Dasug) can you verify with this build? https://github.com/doitsujin/dxvk/actions/runs/26543903902

Yeah, this does appear to fix the issue. 🎉 Comparison screenshots:

Before:

<img width="3840" height="2160" alt="Image" src="https://github.com/user-attachments/assets/0dda3526-17b5-42c9-a92d-4d416a74ea54" />

After: 

<img width="3840" height="2160" alt="Image" src="https://github.com/user-attachments/assets/f1bcf099-00ab-4b58-9586-2ebcf30117bc" />
